plugins/event_source/azure_service_bus.py

```diff
@@ -15,20 +17,52 @@
 
 """
 
+import asyncio
+import concurrent.futures
 import json
+from typing import Any, Dict
 from azure.servicebus import ServiceBusClient
 
 
-def main(queue, args):
+def receive_events(
+    loop: asyncio.AbstractedEventLoop,
+    queue: asyncio.Queue,
+    args: Dict[str, Any]
+):
 
     servicebus_client = ServiceBusClient.from_connection_string(
-        conn_str=args["conn_str"], logging_enable=args.get("logging_enable", True)
+        conn_str=args["conn_str"],
+        logging_enable=args.get("logging_enable", True)
     )
 
     with servicebus_client:
-        receiver = servicebus_client.get_queue_receiver(queue_name=args["queue_name"])
+        receiver = servicebus_client.get_queue_receiver(
+            queue_name=args["queue_name"]
+        )
         with receiver:
             for msg in receiver:
                 data = json.loads(str(msg))
-                queue.put(data)
+                loop.call_soon_threadsafe(queue.put_nowait, data)
                 receiver.complete_message(msg)
+
+
+async def main(queue: asyncio.Queue, args: Dict[str, Any]):
+    loop = asyncio.get_event_loop()
```

Comment:
`get_event_loop` is deprecated, `get_running_loop` is the recommended way: https://docs.python.org/3/library/asyncio-eventloop.html#asyncio.get_event_loop